lib/bson/symbol.rb

```diff
@@ -89,18 +89,73 @@ def to_bson_normalized_key
       to_s
     end
 
+    # Converts this object to a representation directly serializable to
+    # Extended JSON (https://github.com/mongodb/specifications/blob/master/source/extended-json.rst).
+    #
+    # @option options [ true | false ] :relaxed Whether to produce relaxed
+    #   extended JSON representation.
+    #
+    # @return [ Hash ] The extended json representation.
+    def as_extended_json(**options)
+      { "$symbol" => to_s }
+    end
+
+    class Raw
+      def initialize(str_or_sym)
```

Comment:
The methods added here also need docstrings. I was going to  copy them from some other class.